flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -400,6 +400,37 @@ void createIcebergTable(ObjectPath tablePath, CatalogBaseTable table, boolean ig
     }
   }
 
+  /**
+   * schema.equals() contanins getColumns and getWatermarkSpecs
+   * and "primary key"
+   * primary.equals() contains columnNames and constraintName
+   * columnNames are identifierFields of table and constraintName now generate randomly through a function
+   * {@link TableSchema} #primaryKey
+   * so the result is always false because random names are not equal
+   * (two solutions: 1. give specific name like all identifierFields's name combination
+   *                 2. change the equals method because table's metadata has no constraintName)
+   * @param rawSchema Table original schema
+   * @param newSchema Schema after table change
+   * @return Boolean value of whether the schema is consistent
+   */
+  boolean schemaEquals(org.apache.flink.table.api.Schema rawSchema,
```

Comment:
I read the Javadoc comment. When I read the `UnresolvedPrimaryKey` code, it was only comparing the `columnNames`. is it not correct?

```
        @Override
        public boolean equals(Object o) {
            if (this == o) {
                return true;
            }
            if (o == null || getClass() != o.getClass()) {
                return false;
            }
            if (!super.equals(o)) {
                return false;
            }
            UnresolvedPrimaryKey that = (UnresolvedPrimaryKey) o;
            return columnNames.equals(that.columnNames);
        }
```